db/model/Gdoc/GdocBase.ts

```diff
@@ -599,18 +679,27 @@ export class GdocBase extends BaseEntity implements OwidGdocBaseInterface {
         )
 
         const subclassErrors = await this._validateSubclass(this)
-
         this.errors = [...filenameErrors, ...linkErrors, ...subclassErrors]
     }
 
-    toJSON(): Record<string, any> {
+    async loadState(
+        publishedExplorersBySlug: Record<string, any>
+    ): Promise<void> {
+        await this.loadLinkedDocuments()
+        await this.loadImageMetadata()
+        await this.loadLinkedCharts(publishedExplorersBySlug)
+        await this.loadRelatedCharts()
+        await this.validate(publishedExplorersBySlug)
+    }
+
+    toJSON<T extends OwidGdocBaseInterface>(): T {
         return omit(this, [
             "_enrichSubclassContent",
             "_filenameProperties",
             "_getSubclassEnrichedBlocks",
             "_omittableFields",
             "_validateSubclass",
             ...this._omittableFields,
-        ])
+        ]) as any as T
```

Comment:
Yeah I couldn't find a way to do this that will indicate that it will be the interface of the child.

Ideally we can call `someGdocPost.toJson()` and get `OwidGdocPostInterface`